eclipse/src/saros/ui/actions/ChangeXMPPAccountAction.java

```diff
@@ -74,12 +75,23 @@ public ChangeXMPPAccountAction() {
   @Override
   public void run() {
 
+    final XMPPAccount defaultAccount = accountService.getDefaultAccount();
+    final boolean isEmpty = accountService.isEmpty();
+
+    if (defaultAccount == null || isEmpty) {
+      if (!MessageDialog.openQuestion(
+          SWTUtils.getShell(),
+          "Default account missing",
+          "A default account has not been set yet. Do you want set a default account?")) return;
+
+      SWTUtils.runSafeSWTAsync(LOG, this::openPreferences);
```

Comment:
Does this force the user to set a default account? Otherwise, there could still not be a default account requesting it in l. 94, leading to a NPE in XMPPConnectionSupport (see comment in file).

Reply:
You cannot reach line 94 when you enter the if block on line 81. Guess I should make a guard that this is called from SWT anyway so I can remove the asyncCall.

Reply:
Wupsi. My bad. Apparently I just missed that return statement. :wink:

Reply:
Wupsi. My bad. Apparently I just missed that return statement. :wink: